core/src/main/java/jenkins/util/ResourceBundleUtil.java

```diff
@@ -72,14 +77,54 @@ private ResourceBundleUtil() {
             return bundleJSON;
         }
 
-        ResourceBundle bundle = ResourceBundle.getBundle(baseName, locale);
+        ResourceBundle bundle = getBundle(baseName, locale, Jenkins.class.getClassLoader());
+        if (bundle == null) {
+            // Not in Jenkins core. Check the plugins.
+            Jenkins jenkins = Jenkins.getInstance(); // will never return null
+            if (jenkins != null) {
+                for (PluginWrapper plugin : jenkins.getPluginManager().getPlugins()) {
+                    bundle = getBundle(baseName, locale, plugin.classLoader);
+                    if (bundle != null) {
+                        break;
+                    }
+                }
+            }
+        }
+        if (bundle == null) {
+            throw new MissingResourceException("Can't find bundle for base name "
+                    + baseName + ", locale " + locale, baseName + "_" + locale, "");
+        }
 
         bundleJSON = toJSONObject(bundle);
         bundles.put(bundleKey, bundleJSON);
 
         return bundleJSON;
     }
 
+    /**
+     * Get a plugin bundle using the supplied Locale and classLoader
+     *
+     * @param baseName The bundle base name.
+     * @param locale The Locale.
+     * @param classLoader The classLoader
+     * @return The bundle JSON.
+     */
+    private static @CheckForNull ResourceBundle getBundle(@Nonnull String baseName, @Nonnull Locale locale, @Nonnull ClassLoader classLoader) {
+        try {
+            return ResourceBundle.getBundle(baseName, locale, classLoader);
+        } catch (MissingResourceException e) {
+            // fall through and return null.
+            logger.warning(e.getMessage());
```

Comment:
Wait, will this print an average of dozens of messages every time a non-core resource is searched? It iterates over all plugins… If so, this should be changed in master as well.